languages/python/oso/tests/test_polar_roles.py

```diff
@@ -560,10 +547,10 @@ def test_undeclared_role(init_oso):
     """
     oso.load_str(policy)
     with pytest.raises(OsoError):
-        oso.roles.synchronize_data()
+        oso.validate_config()
 
 
-@pytest.mark.skip("TODO: validation")
+@pytest.mark.skip("Can't really test relationship types")
```

Comment:
Without worrying about validating the dot lookup portion of the `parent/2` predicates, wouldn't it be possible for us to validate that there exist a chain of `parent/2` predicates that — if the dot lookups succeed — span from the implied namespace (`"repo"`) to the parent resource (`Org`)? That wouldn't be perfect, but it'd at least offer some level of assurance. Happy to book it as a follow-up if you don't think the payoff would be worth doing at the moment.

Reply:
The way the constraints come out right now there's no way to really know the second arg has to be an "Org", the constraints are on the dot looked up var instead of the output var and we can't assume this rule even uses a dot lookup as they could call an external method or do anything else to get the org.